meta/saiserialize.cpp

```diff
@@ -1131,6 +1135,30 @@ std::string sai_serialize_ip_address_list(
     return sai_serialize_list(list, countOnly, [&](sai_ip_address_t item) { return sai_serialize_ip_address(item);} );
 }
 
+std::string sai_serialize_segment_list(
+        _In_ const sai_segment_list_t& segmentlist,
+        _In_ bool countOnly)
+{
+    SWSS_LOG_ENTER();
+
+    json j;
+    j["count"] = segmentlist.count;
+
+    if(segmentlist.list == NULL || countOnly)
+    {
+      j["list"] = nullptr;
+    }
+    json arr = json::array();
+    for (uint32_t i = 0; i < segmentlist.count; i++)
+    {
+        json item = sai_serialize_ipv6(segmentlist.list[i]);
+        arr.push_back(item);
+    }
+    j["list"] = arr;
```

Comment:
if array is null,  will it still be equal to [], use sai_serialize_list template above